joss/paper.md

```diff
@@ -0,0 +1,105 @@
+
+# Statement of Need
+
+Efficient data stream processing is a foundational challenge in modern computing systems, where the ability to analyze continuous, high-volume data has become more prevalent [@Cordeiro2016]. Applications include sensor networks [@Eiman2003], distributed big-data processing, real-time network traffic analysis, systems log management, fraud monitoring, trading in financial markets, environmental monitoring, and astronomy, which all generate data at rates that exceed practical storage capacity while requiring analysis across varying time horizons.
+
+Downstream addresses limitations of traditional approaches to data streaming by providing algorithms that compress stream history into fixed memory capacities while preserving significant information across various temporal distributions. By exclusively operating with primitive operations and eliminating memory overhead, the framework enables efficient operations in severely resource-constrained environments including embedded systems, where available memory is often measured in kilobytes. For instance, the Cerebras WSE-2 demonstrates this constraint clearly—despite having 40 gigabytes of total on-chip memory, each individual core has access to under 50 kilobytes of memory [@Lie2023].
+
+This capability is particularly valuable for applications such as hereditary stratigraphy in large-scale phylogenetic tracking, where memory-efficient evolutionary history tracking must be maintained across distributed computing environments.
+
+The framework's distinct approach of requiring only single-operation "site selection" with no metadata storage provides significant improvements in space efficiency compared to previous methods. This makes Downstream especially valuable for applications where individual data items may be as small as single bits or bytes—scenarios, where conventional approaches would consume more memory for metadata than for the data itself.
+
+
+# Methods
+
+To simplify the process of stream curation, this approach makes some assumptions: first, the output buffer is of a fixed length; second, stored data remains static without subsequent editing or relocation. Operations on ingested items are therefore limited to discarding without storage, storing, or overwriting previously stored data. This reduces the problem of stream curation to one of site selection—determining the optimal placement of each subsequent data item.
```

Comment:
We should probably define stream curation somewhere, or avoid referencing it as it's not really a standard term

Reply:
optimal might be a bit too strong of a word